keps/sig-api-machinery/4020-unknown-version-interoperability-proxy/README.md

```diff
@@ -151,24 +153,14 @@ API server change:
     with the resources it can serve and receiving a request for a resource
     that is not yet available on that apiserver).
 
-* Discovery merging:
+* Discovery handling:
 
-  - During upgrade or downgrade, it may be the case that no apiserver has a
-    complete list of available resources. To fix the problems mentioned, it's
-    necessary that discovery exactly matches the capability of the system. So,
-    we will use the storage version objects to reconstruct a merged discovery
-    document and serve that in all apiservers.
+  - We will proxy the discovery request to the newest available apiserver in the cluster. In doing so, we will ensure that all discovery requests are handled keeping the future state of the cluster in mind. 
+
+* Version Aware Proxying:
+
+  - With [compatibility versions](https://github.com/kubernetes/enhancements/pull/4395) information available for kubernetes components, we can leverage that to perform version aware proxying between API servers, and ensure that a request is proxied to an eligible apiserver given that it was generated by a client compatible with the apiserver/s present in a cluster. Version aware proxying lets us ensure that requests from clients (possibly) expecting to use a feature introduced in later K8s versions,  are not incorrectly routed to apiservers at older versions.
```

Comment:
Do we need to describe how clients would declare their version to servers here?

Reply:
Added a suggestion for specifying this information in client request headers. Also added a TODO section to flesh out the details of the same in Design Details.